plasmapy/particles/particle_collections.py

```diff
@@ -294,6 +294,65 @@ def symbols(self) -> List[str]:
         """A `list` of the symbols of the particles."""
         return self._get_particle_attribute("symbol")
 
+    def average_particle(
+        self,
+        abundances=None,
+        *,
+        use_rms_charge: bool = False,
+        use_rms_mass: bool = False,
+    ) -> Union[CustomParticle, Particle]:
+        """
+        Return a particle with the average mass and charge.
+
+        By default, the mean will be used as the average. If the ``abundances``
+        are provided, then this method will return the weighted mean. If
+        ``use_rms_charge`` or ``use_rms_mass`` is `True`, then return the root
+        mean square of the charge or mass, respectively. If all items in the
+        |ParticleList| are the same, then return that item.
+
+        Parameters
+        ----------
+        abundances : array-like, optional
```

Comment:
Here's I'm matching NumPy's use and definition of [array-like](https://numpy.org/doc/stable/glossary.html#term-array_like).  In #1149 and #1158, I've been hoping to set up a glossary and then make it so that terms like this automatically get linked to the appropriate terms in the glossary (or in the glossary of another package) in the rendered docs.  I've made a good start to the glossary but haven't figured out how to do the auto-linking in the same way that Astropy has, so that part is a work in progress.

Reply:
The glossary is a great idea! 

My concern is just that the following is technically an `np.ndarray'

```python
np.array(['a', 'b'])

```
But this wouldn't be valid input for the abundances keyword.

Reply:
Yep, that makes sense.  I modified the description for `abundances` to say that they have to be real numbers.